sessionctx/variable/sysvar_test.go

```diff
@@ -506,12 +506,16 @@ func (*testSysVarSuite) TestInstanceScopedVars(c *C) {
 	c.Assert(err, IsNil)
 	c.Assert(val, Equals, fmt.Sprintf("%g", MemoryUsageAlarmRatio.Load()))
 
-	val, err = GetSessionOrGlobalSystemVar(vars, TiDBConfig)
-	c.Assert(err, IsNil)
-	conf := config.GetGlobalConfig()
-	j, err := json.MarshalIndent(conf, "", "\t")
-	c.Assert(err, IsNil)
-	c.Assert(val, Equals, config.HideConfig(string(j)))
+	// Because JSON key order is non-deterministic, this test is flaky.
```

Comment:
Maybe try to unmarshal the val into config?

Reply:
I figured out what causes it. It was not json key order, but a test in `server/http_handler_test.go` which changed the expected value and did not restore it back.